drivers/ethernet/eth_stm32_hal.c

```diff
@@ -998,13 +1006,13 @@ static int eth_initialize(const struct device *dev)
 	tx_config.CRCPadCtrl = ETH_CRC_PAD_INSERT;
 #endif /* CONFIG_ETH_STM32_HAL_API_V2 */
 
-	dev_data->link_up = false;
+	eth_dev_data->link_up = false;
```

Comment:
Please do not rename variables with no reason for it.

Reply:
This is corrected now.